sections/builder.js

```diff
@@ -2282,5 +2282,75 @@ export default [
         content: "Calling `queryContext` with no arguments will return any context configured for the query builder instance."
       }
     ]
-  }
+  },
+  {
+    type: "heading",
+    size: "md",
+    content: "Extending Query Builder",
+    href: "Builder-extending"
+  },
+  {
+    type: "text",
+    content: [
+      "**Important:** this feature is experimental and it's API may change in the future.",
+      "It allows to add custom function the the Query Builder. It should be invoked only once in the app.",
+      "Example:"
+    ]
+  },
+  {
+    type: "code",
+    language: "js",
+    content: `
+      const Knex = require('knex');
+      Knex.QueryBuilder.extend('customSelect', function(value) {
+        return this.select(this.client.raw(\`\${value} as value\`));
+      });
+      
+      const meaningOfLife = await knex('accounts')
+        .customSelect(42);
+    `
+  },
+  {
+    type: "text",
+    content: [
+      "If using TypeScript, you can extend the QueryBuilder interface with your custom method.",
+      "1. Create a `knex.d.ts` file inside a `@types` folder.",
```

Comment:
couldn't it be any file inside any folder?

Reply:
It can be inside any other folder, just for example sake I need to reference to the same folder in bullet 2